Remove compilation of i2c for arm split

zvecr opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
Another little common slice from #7072 and #7987.

As #6494 is still on the back burner, this removes the need to have stub files, and i2c enabled in the chibios conf files, for it to just be thrown away....

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

tzarc commented:
Should we start to think about having "driver enable flags" or something where we have something like:

```
if "feature enabled that requires i2c"
    set flag "compile i2c"
endif

if flag is set "compile i2c"
    QUANTUM_LIB_SRC += i2c_master.c
endif
```

...or something to that effect?

I'm just worried that we're getting to the stage where any sort of refactoring will require a bunch of duplicated changes because AVR split serial needs i2c, or you're using an OLED, or the i2c-based ws2812 driver is turned on, etc. etc.

zvecr commented:
The main issue I am trying to avoid with this PR, is that if `i2c_master` is included, it needs to be enabled and configured within HAL. This makes stuff like CTPC throw errors right now, as the default files assume that if you are compiling `i2c_master`, you kinda want I2C. In this case, we have no slave code, and are not targeting a mixed `proton c + pro micro ` setup.

The "Feature X requires Driver Y" pattern is already starting to form, you can already see it with rgb(light|matrix) and ws2812. However in this particular use case, split uses defines to pick the transport, so right now you would still end up with a dirty "if arm" thrown in there.

I have added a card to the backlog (<https://github.com/qmk/qmk_firmware/projects/2>) for some future investigation work (which I imagine will start once this endless MMF stuff gets finished).
